signal_interpolation_fourier.py

```diff
@@ -0,0 +1,589 @@
+# Module for Fourier interpolation of pulsed signals along simulated radio footprints of cosmic-ray air showers
+# Author: A. Corstanje (a.corstanje@astro.ru.nl), 2023
+#
+# See article: arXiv:xxxx.xxxxx, doi xxx
+# Please cite this when using code and/or methods in your analysis
+
+import numpy as np
+from scipy.signal import hilbert, resample
+import interpolation_fourier as interpF
+
+class interp2d_signal:
+
+    def get_spectra(self, signals, verbose=False):
+        """
+        # Do FFT of 'signals', assumed shape (Nants, Nsamples, Npol) i.e. the time traces are along the second axis.
+        # Produce absolute-amplitude spectrum and phase spectrum
+        """
+        Nsamples = signals.shape[1]
+
+        if verbose: print('Doing FFTs...', end=' ')
+        all_antennas_spectrum = np.fft.rfft(signals, axis=1)
+        abs_spectrum = np.abs(all_antennas_spectrum)
+        phasespectrum = np.angle(all_antennas_spectrum)
+        unwrapped_phases = np.unwrap(phasespectrum, axis=1, discont=0.7*np.pi)
+        if verbose: print('done.')
+
+        #antnr = 1555
+        #print(pos_sim_UVW[antnr])
+        freqs = np.fft.rfftfreq(Nsamples, d=0.1e-9)
+        freqs /= 1.0e6 # in MHz
+
+        freqstep = freqs[1] - freqs[0]
+        nof_freq_channels = len(np.where( (freqs < 500) )[0])
+        if verbose: print('Frequency channel width %2.4f MHz, there are %d channels between 0 and 500 MHz' % (freqstep, nof_freq_channels))
+
+        return (freqs, all_antennas_spectrum, abs_spectrum, phasespectrum, unwrapped_phases)
+
+    def hilbert_envelope_timing(self, signals, lowfreq=30.0, highfreq=500.0, upsample_factor=10, sum_over_pol=True, do_hilbert_envelope=True):
+        """
+        Produce pulse arrival times from Hilbert envelope maxima (if do_hilbert_envelope) or from raw E-field maxima.
+        Assumed shape for 'signals' is (Nant, Nsamples, Npols), i.e. time traces along second axis
+        Filtering is done between lowfreq and highfreq in MHz, timing is done on filtered signals (after inverse-FFT)
+        Option sum_over_pol is to sum the square of the Hilbert envelopes of each polarization, to get one arrival time over all polarizations (may help at low signal strength), default True.
+
+        Parameters
+        ----------
+        signals : 3D array of shape (Nant, Nsamples, Npols)
+        lowfreq : low-frequency cutoff, default 30.0 MHz
+        highfreq : high-frequency cutoff, default 500.0 MHz
+        upsample_factor : upsampling factor to use for sub-sample timing accuracy, default 10
+        sum_over_pol : sum over polarizations if do_hilbert_envelope, default True
+        do_hilbert_envelope : use Hilbert envelope for timing if True, else direct E-field maximum
+        """
+        (Nant, Nsamples, Npols) = signals.shape
+
+        if self.verbose: print('Bandpass filtering %d to %d MHz' % (int(lowfreq), int(highfreq)))
```

Comment:
Can you add a 'return' for clarity? In NuRadioMC we all use a code linter that fixes these things for us. But it is a bit strict, so let's not start on this just now.

Reply:
Also there is no local variable verbose, no? (it is getting late, don't mind me. )

Reply:
The self.verbose may be a bit counter intuitive, since the class isn't initialized with it. Anyway, works if need be.

Reply:
A return? A print statement adds an end-of-line automatically, if that's what you mean.
Or do you want a \n added to every printed string?

Reply:
I meant use the formatting, instead of having it in one line. 
if self.verbose:
    print("Bla")

Reply:
Oh allright, of course.
This will add a lot of (mostly empty) lines to the code though, which I tried to avoid.
But can do.

Reply:
Done.